tests/integration/test_exec_disconnect.py

```
@@ -0,0 +1,88 @@
+    """Killing the SSH client mid-command terminates the remote process.
+
+    Runs a long silent `sleep` over the exec channel, records its PID (via a
+    file the command writes), kills the SSH client to simulate a disconnect,
+    then polls `/proc/<pid>` from a fresh exec until the process is gone. The
+    silent command forces detection through the agent's keepalive probe rather
+    than an I/O error, on both the non-PTY and PTY paths.
+    """
+    shed_server.create(test_shed_name, image="base")
+
+    label = "pty" if pty else "nopty"
+    pid_file = f"/tmp/shed-disconnect-{label}.pid"
```

Ruff (0.15.20)

[error] Probable insecure usage of temporary file or directory: "/tmp/shed-disconnect-"

(S108)